# Graded assignment 1

Mark Klik & Misja Mikkers


# Packages

We need the package `tidyverse`

In [5]:
library(tidyverse) 

# Assignment 1

First, we generate some random data:

In [6]:
df1 <- as.data.frame(matrix(runif(20*2), ncol=20) )
head(df1)

V1        V2        V3        V4        V5        V6         V7       
1 0.8710647 0.9779129 0.9320461 0.9375003 0.1702073 0.08857130 0.9842537
2 0.3252264 0.2943260 0.7312783 0.1453897 0.5223858 0.07483392 0.6125601
  V8        V9        V10       V11       V12       V13        V14         
1 0.6103444 0.2037647 0.3673268 0.9604186 0.9987291 0.01536995 0.6473011510
2 0.7184080 0.1312560 0.7348230 0.6408337 0.7039797 0.57191716 0.0009656565
  V15       V16       V17       V18       V19       V20      
1 0.2326299 0.8051571 0.7080144 0.3602606 0.2065807 0.2295829
2 0.4872314 0.1015781 0.1553903 0.7565632 0.4612835 0.7778022

_df1_ is a wide dataframe. The goal of the assignment is to change _df1_ into a long dataframe _df2_. And show the first rows of the dataframe using `head(df2)`

The dataframe _df2_ should look like this:

| Key | Value      |
|-----|------------|
| V1  | 0.09413864 |
| V1  | 0.69782526 |
| V2  | 0.23460454 |
| V2  | 0.51520514 |
| V3  | 0.63181844 |
| V3  | 0.22807211 |

In [7]:
df2 <- df1 %>%
  

gather(Key,Value)
head(df2)

Key Value    
1 V1  0.8710647
2 V1  0.3252264
3 V2  0.9779129
4 V2  0.2943260
5 V3  0.9320461
6 V3  0.7312783

## Assignment 2

We will generate some data:

In [8]:
df3 <- as.data.frame(read.table(header=TRUE, text='
      id        V_1       V_2                
          1  42.467573  -87.810047     
          2  42.049148  -88.273029
          3  39.110539  -90.324080
                       '))

head(df3)

id V_1      V_2      
1 1  42.46757 -87.81005
2 2  42.04915 -88.27303
3 3  39.11054 -90.32408

In [9]:
df4 <- as.data.frame(read.table(header=TRUE, text='
     V_3            V_4         id                 
      one         1h            1
      two         1h            1
      three       8h            2
      four        1h            4
'))

head(df4)

V_3   V_4 id
1 one   1h  1 
2 two   1h  1 
3 three 8h  2 
4 four  1h  4


You will have to merge these files, returning all rows and all columns from both _df3_ and _df5_. Where there are not matching values, return _NA_ for the missings. 

This should be the end result:

| id | V_1      | V_2       | V_3   | V_4 |
|----|----------|-----------|-------|-----|
| 1  | 42.46757 | -87.81005 | one   | 1h  |
| 1  | 42.46757 | -87.81005 | two   | 1h  |
| 2  | 42.04915 | -88.27303 | three | 8h  |
| 3  | 39.11054 | -90.32408 | NA    | NA  |
| 4  | NA       | NA        | four  | 1h  |



In [11]:
df5 <- merge(df3, df4, all="true")
df5

id V_1      V_2       V_3   V_4
1 1  42.46757 -87.81005 one   1h 
2 1  42.46757 -87.81005 two   1h 
3 2  42.04915 -88.27303 three 8h 
4 3  39.11054 -90.32408 NA    NA 
5 4        NA        NA four  1h

End of notebook